### PR TITLE
Chapter 9: Advanced login

### PR DESCRIPTION
**Pull request by phantuananh1206**
Chapter 9: Advanced login

### CODE REVIEW

**Comment by phantuananh1206** [comment]
image

**Comment by phantuananh1206** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -11,12 +12,35 @@ class User < ApplicationRecord
...
+    self.remember_token = User.new_token
+    update_column :remember_digest, User.digest(remember_token)
+  end
+
+  def authenticated? remember_token
+    if remember_digest.nil?
+      false
+    else
+      BCrypt::Password.new(remember_digest).is_password?(remember_token)
+    end
```
Comment (vanvtt-0952) [inline app/models/user.rb:40]: thay code if else thành `return flase unless remember_digest # if remember_degest.nil?`
`BCrypt::Password.new(remember_digest).is_password?(remember_token)`

**Comment by phantuananh1206** [comment]
ready

**Comment by vanvtt-0952** [comment]
merged

**Pull request merged**